### PR TITLE
Change interface of waitanyinvoice to use pay_index instead of label

### PR DESCRIPTION
**Pull request by ZmnSCPxj**
Dependent on #445 

This is idea of @rustyrussell, to use pay_index in waitanyinvoice.

### CODE REVIEW

**Comment by ZmnSCPxj** [comment]
Rebased on top of master.

**Comment by ZmnSCPxj** [comment]
Rebased again.

File: wallet/wallet.c [hunk 1]
```diff
@@ -1297,7 +1297,8 @@ bool wallet_invoices_load(struct wallet *wallet, struct invoices *invs)
 	int count = 0;
 	sqlite3_stmt *stmt = db_query(__func__, wallet->db,
 				"SELECT id, state, payment_key, payment_hash, "
-				"label, msatoshi, expiry_time FROM invoices;");
+				"label, msatoshi, expiry_time, COALESCE(pay_index,0) "
```
Comment (rustyrussell) [inline wallet/wallet.c]: COALESCE() is unnecessary: the `inv->pay_index = sqlite3_column_int64(stmt, 7);` above will give 0 as you expect if the field is nul (I just re-checked the docs).

File: lightningd/invoice.c [hunk 2]
```diff
@@ -83,6 +83,7 @@ static void tell_waiter(struct command *cmd, const struct invoice *paid)
 	json_add_hex(response, "rhash", &paid->rhash, sizeof(paid->rhash));
 	json_add_u64(response, "msatoshi", paid->msatoshi);
 	json_add_bool(response, "complete", paid->state == PAID);
+	json_add_u64(response, "pay_index", paid->pay_index);
```
Comment (rustyrussell) [inline lightningd/invoice.c]: Make this field conditional on 'complete' being true.  Otherwise it's just weird...

**Comment by ZmnSCPxj** [comment]
Modified as requested.

**Comment by rustyrussell** [comment]
Rebased to fix trivial conflict; waiting for travis before final merge.  Thanks!

**Comment by ZmnSCPxj** [comment]
Some modifications to the `test_waitanyinvoice_reverse` were not propagated in the rebase, will modify.

**Pull request merged**